File: tests/transpiler/x/fs/bench_block.fs
// Generated 2025-07-24 19:13 +0000

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
open System

let __memStart = System.GC.GetTotalMemory(false)
let __start = _now()
let n: int = 1000
let mutable s: int = 0
for i in 1 .. (n - 1) do
    s <- s + i
let __end = _now()
let __memEnd = System.GC.GetTotalMemory(false)
let __dur_us = (__end - __start) / 1000
let __mem_diff = __memEnd - __memStart
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"simple\"\n}" __dur_us __mem_diff

File: tests/transpiler/x/pas/bench_block.pas
{$mode objfpc}
program Main;
uses SysUtils;
var _nowSeed: int64 = 0;
var _nowSeeded: boolean = false;
procedure init_now();
var s: string; v: int64;
begin
  s := GetEnvironmentVariable('MOCHI_NOW_SEED');
  if s <> '' then begin
    Val(s, v);
    _nowSeed := v;
    _nowSeeded := true;
  end;
end;
function _now(): integer;
begin
  if _nowSeeded then begin
    _nowSeed := (_nowSeed * 1664525 + 1013904223) mod 2147483647;
    _now := _nowSeed;
  end else begin
    _now := Integer(GetTickCount64());
  end;
end;
var
  bench_start_0: integer;
  bench_dur_0: integer;
  n: integer;
  s: integer;
  i: integer;
begin
  init_now();
  bench_start_0 := _now();
  n := 1000;
  s := 0;
  for i := 1 to (n - 1) do begin
  s := s + i;
end;
  bench_dur_0 := (_now() - bench_start_0) div 1000;
  writeln('{');
  writeln(('  "duration_us": ' + IntToStr(bench_dur_0)) + ',');
  writeln('  "memory_bytes": 0,');
  writeln(('  "name": "' + 'simple') + '"');
  writeln('}');
end.
